### PR TITLE
Also cleanup custom target paths for not installed not default targets

### PR DESCRIPTION
**Pull request by thiblahute**

### CODE REVIEW

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -980,8 +981,11 @@ def first_supported_argument_method(self, args, kwargs):
         mlog.log('First supported argument:', mlog.red('None'))
         return []
 
-class ModuleState:
-    pass
+ModuleState = namedtuple('ModuleState', [
```
Comment (nirbheek) [inline mesonbuild/interpreter.py]: This change seems unrelated to the rest of the changes. Did you add this on purpose?

Comment (thiblahute) [inline mesonbuild/interpreter.py]: Well, this is another commit actually.

Comment (nirbheek) [inline mesonbuild/interpreter.py]: I asked because I was confused what that change had to do with cleaning up files properly. It doesn't make any difference to have it though, so it's up to Jussi :)

Comment (thiblahute) [inline mesonbuild/interpreter.py]: RIght just past by that part of the code and thought it was worth making that change :)
